deps/verifier/src/cca/mod.rs

```diff
@@ -79,71 +84,49 @@ impl Verifier for CCA {
+            std::env::var(CCA_CONFIG_FILE).unwrap_or_else(|_| DEFAULT_CCA_CONFIG.to_string());
+
+        let config = Config::try_from(Path::new(&config_file))
+            .map_err(|e| anyhow!("parsing {config_file}: {e}"))?;
+
         let ReportData::Value(expected_report_data) = expected_report_data else {
             bail!("CCA verifier must provide report data field!");
         };
 
         let expected_report_data = regularize_data(expected_report_data, 64, "REPORT_DATA", "CCA");
 
-        let evidence = serde_json::from_slice::<CcaEvidence>(evidence)
+        let evidence = unwrap_evidence(evidence)?;
+        let evidence = serde_json::from_str::<CcaEvidence>(&evidence)
             .context("Deserialize CCA Evidence failed.")?;
 
-        let host_url =
-            std::env::var(VERAISON_ADDR).unwrap_or_else(|_| DEFAULT_VERAISON_ADDR.to_string());
-
-        let discovery = Discovery::from_base_url(format!("http://{:}", host_url))?;
-
-        let verification_api = discovery.get_verification_api().await?;
-
-        let relative_endpoint = verification_api
-            .get_api_endpoint("newChallengeResponseSession")
-            .context("Failed to discover the verification endpoint details.")?;
-
-        let api_endpoint = format!("http://{:}{}", host_url, relative_endpoint);
-
-        // create a ChallengeResponse object
-        let cr = ChallengeResponseBuilder::new()
-            .with_new_session_url(api_endpoint)
-            .build()?;
-
-        let token = evidence.token;
-        let n = Nonce::Value(expected_report_data.clone());
-        let result = match cr.run(n, my_evidence_builder, token.clone()).await {
-            Err(e) => {
-                error!("Error: {}", e);
-                bail!("CCA Attestation failed with error: {:?}", e);
+        let ear: Ear = match config.cca_verifier {
+            config::CcaVerifierConfig::Remote { .. } => {
+                remote::verify(config, &evidence.token, &expected_report_data).await?
+            }
+            config::CcaVerifierConfig::Local { .. } => {
+                local::verify(config, &evidence.token, &expected_report_data)?
             }
-            Ok(attestation_result) => attestation_result,
         };
 
-        // Get back the pub key to decrypt the ear which holds raw evidence and the session nonce
-        let public_key_pem = verification_api.ear_verification_key_as_pem()?;
-        let dk = jwt::DecodingKey::from_ec_pem(public_key_pem.as_bytes())
-            .context("get the decoding key from the pem public key")?;
-        let plain_ear = Ear::from_jwt(result.as_str(), jwt::Algorithm::ES256, &dk)
-            .context("decrypt the ear with the decoding key")?;
-
-        let ear_nonce = plain_ear.nonce.context("get nonce from ear")?;
-        let nonce_byte = base64::engine::general_purpose::STANDARD
-            .decode(ear_nonce.to_string())
-            .context("decode nonce byte from ear")?;
+        info!("EAR: {:?}", ear);
```

Comment:
Could we mark this log `debug` level, or add some context if we want to keep `info!`